feature/experimental/bgp/ate_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go

```diff
@@ -0,0 +1,449 @@
+	if localAS != nbr.localAS {
+		t.Errorf("BGP localAS: got %v, want %v", localAS, nbr.localAS)
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != nbr.localAS {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+
+	verifyPrefixesTelemetry(t, dut, 3, nbr.isV4)
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string, prefixes []string) *ondatra.ATETopology {
+	t.Helper()
+	t.Log("Configure ATE interface")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+	iDut1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
+
+	bgpDut1 := iDut1.BGP()
+	peer := bgpDut1.AddPeer().WithPeerAddress(ateParams.peerIP).WithLocalASN(ateParams.localAS)
+
+	if connectionType == connInternal {
+		peer.WithTypeInternal()
+	} else {
+		peer.WithTypeExternal()
+	}
+
+	network1 := iDut1.AddNetwork("bgpNeti1")
+	network2 := iDut1.AddNetwork("bgpNeti2")
+	network3 := iDut1.AddNetwork("bgpNeti3")
+
+	if ateParams.isV4 {
+		network1.IPv4().WithAddress(prefixes[0]).WithCount(1)
+		network1.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 4400, 3300)
+
+		network2.IPv4().WithAddress(prefixes[1]).WithCount(1)
+		network2.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 7700)
+		network2.BGP().Communities().WithPrivateCommunities("200:1")
+
+		network3.IPv4().WithAddress(prefixes[2]).WithCount(1)
+		network3.BGP().WithNextHopAddress(ateSrc.IPv4)
+	} else {
+		network1.IPv6().WithAddress(prefixes[0]).WithCount(1)
+		network1.BGP().WithNextHopAddress(ateSrc.IPv6).AddASPathSegment(55000, 4400, 3300)
+
+		network2.IPv6().WithAddress(prefixes[1]).WithCount(1)
+		network2.BGP().WithNextHopAddress(ateSrc.IPv6).AddASPathSegment(55000, 7700)
+		network2.BGP().Communities().WithPrivateCommunities("200:1")
+
+		network3.IPv6().WithAddress(prefixes[2]).WithCount(1)
+		network3.BGP().WithNextHopAddress(ateSrc.IPv6)
+	}
+	return topo
+}
+
+func applyBgpPolicy(policyName string, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	niProto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := niProto.GetOrCreateBgp()
+
+	pg := bgp.GetOrCreatePeerGroup("ATE")
+	pg.PeerGroupName = ygot.String("ATE")
+
+	pg.GetOrCreateApplyPolicy().ImportPolicy = []string{policyName}
+
+	return niProto
+}
+
+func createBgpNeighbor(nbr *bgpNbr, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	niProto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := niProto.GetOrCreateBgp()
+
+	global := bgp.GetOrCreateGlobal()
```

Comment:
Could you enable Global IPv4/IPv6 AFI as well here ?

global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)

Reply:
I would recommend Cisco to address this comment in a separate Pull. Because of the following,

As per the [yang definition](https://github.com/openconfig/public/blob/85a6f6ef5216fad43187ea60d0fe6722b855b293/release/models/bgp/openconfig-bgp-common-multiprotocol.yang#L198C1-L208), the default is "disabled". Even though there are discussions for making global/ipv4-unicast and global/ipv6-unicast enabled as a default in pull/815, the Pull is still under discussions. Given the current situation, we may need further discussions on this topic to decide if this request needs to be addressed by vendor specific deviation or we just consider sending global/afi-safi==enabled for ipv4-unicast and ipv6-unicast.

We have future tests in the pipeline already https://github.com/openconfig/featureprofiles/pull/1776 to check specifically on this requirement for compliance w/ the yang definitions w/ expectations that the non-compliant implementations would have a deviation.

For viz: @dplore for further clarifications as he was actively involved on this topic.

Reply:
Yes by default Global AFI/SAFI is Disabled/False for cisco, so we need to set it to true to enable them at bgp global.
Same is done in all of the other BGP tests.

Reply:
Global enable should be ok.  However, we should be able to enable at the neighbor level and leave the global enable as-is.  Is that supported by Cisco?

Reply:
Well if I understand the ask here clearly, it is that, even though the test is enabling the afi/safi for IPv4 and IPv6 under the neighbor hierarchy, ask is to enable it under the global AFI/SAFI as well? This is irrespective of the fact that the yang states that the default == false https://github.com/openconfig/public/blob/master/release/models/bgp/openconfig-bgp-common-multiprotocol.yang#L217 at the global level. I know we (Google) internally decided that while pushing configs we will ensure that the global/ipv4-unicast is updated to enabled == true always. However, this arrangement doesn't exist for global/ipv6-unicast. Seems some implementations require that global/ipv6-unicast should also be == enabled and just having neighbor/ipv6-unicast == enabled wont work? This is an agreement we need to make internally.

Reply:
So post internal discussions, lets enable IPv4-unicast and IPv6-unicast at the global level as well while we are still working on deciding what should be the long term state here. 

@s-nandu, can you please accommodate the change?

Reply:
I'd prefer to see the test only enable IPv6 at the neighbor level.

Cisco may submit a PR to add a deviation to this test which also enables IPv6 at the global level.

Reply:
@dplore based on our offline conversation, seems we had an agreement to not explicitly enable both v4-unicast and v6-unicast at the global level and all implementations that require these to be explicitly enabled even though these address-families are already enabled either at the neighbor or at the peer-group level need to have 2 different deviations, one for enabling v4 at the global level and another for enabling v6 at the global level. Is your understanding the same as mine?

Reply:
> 

@dplore re: ipv6 enabled at the neighbor, but disabled globally, what is the goal of this test? IOS XR in particular forbids configuration of an afi on a neighbor if that afi is not enabled globally. It seems that this would be a misconfiguration, no?

Reply:
Well, as per the yang definition https://github.com/openconfig/public/blob/master/release/models/bgp/openconfig-bgp-common-multiprotocol.yang#L217, default for Global AFI/SAFI is enabled=false.

Intent is that, if this boolean is set at the neighbor/peer-group level, that should be the config. If not defined at the peer-group/neighbor level then the setting in the global level should take effect, which btw has a default of AFI/SAFI disabled.